# 01 — Inspeção inicial dos dados

Este notebook integra o Projeto 001 — Renda e Longevidade, que
investiga indicadores econômicos e demográficos dos países da
América do Sul entre 2000 e 2023, com referência global e
aprofundamento em Brasil, Argentina, Uruguai e Paraguai.

## Objetivo

Inspecionar a estrutura dos arquivos de PIB per capita em paridade
de poder de compra, expectativa de vida ao nascer e população total,
obtidos no World Development Indicators, do Banco Mundial.

Esta etapa busca compreender a organização dos arquivos, identificar
os períodos disponíveis e observar a representação de valores
ausentes, preparando as próximas etapas de validação e integração.

As observações serão registradas ao longo da inspeção, com base
nas evidências apresentadas neste notebook.

In [1]:
# imports
import sys
import pandas as pd

pd.set_option("display.max_columns", None)

print(
    f"Python - {sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
)
print(f"Pandas - {pd.__version__}")

Python - 3.13.14
Pandas - 3.0.5


## Leitura dos arquivos

Nesta etapa, os três arquivos principais são carregados para
inspeção, preservando todas as colunas disponíveis.

Uma amostra inicial de cada conjunto permite observar sua
organização e orientar a investigação da estrutura nas etapas
seguintes.

In [2]:
# df1 - GDP per Capita
df_1 = pd.read_csv(
    "../data/extracted/API_NY.GDP.PCAP.PP.KD_DS2_en_csv_v2_349558.csv", skiprows=4
)
df_1.head(1)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,1970,1971,1972,1973,1974,1975,1976,1977,1978,1979,1980,1981,1982,1983,1984,1985,1986,1987,1988,1989,1990,1991,1992,1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025,Unnamed: 70
0,Aruba,ABW,"GDP per capita, PPP (constant 2021 internation...",NY.GDP.PCAP.PP.KD,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,33146.670548,34079.230463,34458.142298,34627.743475,35580.533978,35476.409987,34506.53308,35534.274503,35361.165889,35316.62294,37618.992616,38827.467274,38195.317608,38182.392526,40101.787613,38926.475,38664.39951,39562.448848,39982.511992,35074.684122,34037.542321,34926.100892,34058.310233,35711.929051,35837.324093,36753.404173,36926.113069,38213.289239,39378.228411,39754.560072,30408.82709,35207.170909,39083.46164,42783.248907,45428.002728,NaN,NaN


In [3]:
# df_2 - Expectativa de Vida ao Nascer
df_2 = pd.read_csv(
    "../data/extracted/API_SP.DYN.LE00.IN_DS2_en_csv_v2_330385.csv", skiprows=4
)
df_2.head(1)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,1970,1971,1972,1973,1974,1975,1976,1977,1978,1979,1980,1981,1982,1983,1984,1985,1986,1987,1988,1989,1990,1991,1992,1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025,Unnamed: 70
0,Aruba,ABW,"Life expectancy at birth, total (years)",SP.DYN.LE00.IN,64.049,64.215,64.602,64.944,65.303,65.615,66.126,66.385,66.744,67.098,67.44,67.783,68.335,68.862,69.278,69.564,69.808,70.054,70.271,70.507,70.771,71.344,71.485,71.606,71.711,71.792,71.831,72.448,72.519,72.531,72.546,72.592,72.717,72.777,72.796,72.832,72.856,72.904,72.94,72.857,72.939,73.044,73.135,73.236,73.223,73.415,73.498,73.642,73.783,73.974,74.297,74.578,74.841,75.072,75.261,75.405,75.54,75.62,75.88,76.019,75.406,73.655,76.226,76.353,76.5,NaN,NaN


In [4]:
# df_3 - População
df_3 = pd.read_csv(
    "../data/extracted/API_SP.POP.TOTL_DS2_en_csv_v2_350115.csv", skiprows=4
)
df_3.head(1)

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,1970,1971,1972,1973,1974,1975,1976,1977,1978,1979,1980,1981,1982,1983,1984,1985,1986,1987,1988,1989,1990,1991,1992,1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025,Unnamed: 70
0,Aruba,ABW,"Population, total",SP.POP.TOTL,54922.0,55578.0,56320.0,57002.0,57619.0,58190.0,58694.0,58990.0,59069.0,59052.0,58950.0,58781.0,58047.0,58299.0,58349.0,58295.0,58368.0,58580.0,58776.0,59191.0,59909.0,60563.0,61276.0,62228.0,62901.0,61728.0,59931.0,59159.0,59331.0,60443.0,62753.0,65896.0,69005.0,73685.0,77595.0,79805.0,83021.0,86301.0,88451.0,89659.0,90588.0,91439.0,92074.0,93128.0,95138.0,97635.0,99405.0,100150.0,100917.0,101604.0,101838.0,102591.0,104110.0,105675.0,106807.0,107906.0,108727.0,108735.0,108908.0,109203.0,108587.0,107700.0,107310.0,107359.0,107995.0,108785.0,NaN


## Estrutura e dimensões

As amostras apresentam colunas de identificação da economia e do
indicador, seguidas por colunas anuais.

A seguir, são verificadas as quantidades de linhas e colunas dos
três conjuntos.

In [5]:
listas_dfs = [df_1, df_2, df_3]
indicadores = ["GDP per Capita", "Expectativa de Vida ao Nascer", "População Total"]
for i in range(3):
    linhas, colunas = listas_dfs[i].shape
    print(f"\nDataframe: {indicadores[i]}")
    print(f"  - Nº Linhas: {linhas}")
    print(f"  - Nº Colunas: {colunas}")


Dataframe: GDP per Capita
  - Nº Linhas: 265
  - Nº Colunas: 71

Dataframe: Expectativa de Vida ao Nascer
  - Nº Linhas: 265
  - Nº Colunas: 71

Dataframe: População Total
  - Nº Linhas: 265
  - Nº Colunas: 71


Os três DataFrames apresentam 265 linhas e 71 colunas.

A igualdade das dimensões sugere uma organização semelhante,
mas não comprova que os conjuntos contenham os mesmos códigos
ou as mesmas colunas.

## Período apresentado nos arquivos

Os arquivos apresentam colunas anuais de 1960 a 2025.
A presença dessas colunas não significa que todos os registros
possuam valores disponíveis em todo o intervalo.

Nesta inspeção inicial, o período completo dos arquivos é mantido.
O recorte analítico do projeto permanece entre 2000 e 2023.

## Inspeção inicial de valores ausentes

A verificação a seguir considera as linhas cujos códigos pertencem
à lista de países sul-americanos definida para esta inspeção.

São observadas todas as colunas disponíveis, sem aplicar ainda
o recorte temporal da análise.

In [6]:
AMS = "ARG BOL BRA CHL COL ECU GUY PRY PER SUR URY VEN".split()

In [7]:
df_1[df_1["Country Code"].isin(AMS)].isnull().sum()

Country Name       0
Country Code       0
Indicator Name     0
Indicator Code     0
1960              12
                  ..
2022               1
2023               1
2024               1
2025               1
Unnamed: 70       12
Length: 71, dtype: int64

In [8]:
df_2[df_2["Country Code"].isin(AMS)].isnull().sum()

Country Name       0
Country Code       0
Indicator Name     0
Indicator Code     0
1960               0
                  ..
2022               0
2023               0
2024               0
2025              12
Unnamed: 70       12
Length: 71, dtype: int64

In [9]:
df_3[df_3["Country Code"].isin(AMS)].isnull().sum()

Country Name       0
Country Code       0
Indicator Name     0
Indicator Code     0
1960               0
                  ..
2022               0
2023               0
2024               0
2025               0
Unnamed: 70       12
Length: 71, dtype: int64

Nos subconjuntos sul-americanos dos três DataFrames, foram
identificados valores ausentes. A inspeção considera todas as
colunas disponíveis, incluindo anos anteriores ao recorte do estudo.

A coluna `Unnamed: 70` decorre de um campo adicional sem nome,
produzido pelo delimitador final do CSV, e apresenta valores
ausentes. Ela deve ser distinguida das colunas anuais na
interpretação das contagens.

A verificação passou a apresentar a quantidade de valores
ausentes por coluna. A avaliação específica do período de
2000 a 2023 será realizada posteriormente.

## Interpretação da unidade de registro

Cada linha dos arquivos reúne a identificação de uma economia
ou agregado, o nome e o código do indicador e uma série temporal
organizada em colunas anuais de 1960 a 2025.

Nos subconjuntos sul-americanos selecionados, cada linha corresponde
a um país e ao indicador do respectivo arquivo. Os campos anuais
vazios são representados como `NaN` após a leitura pelo Pandas.

Para localizar uma observação país-ano, identifico o país pelo
seu código e seleciono a coluna do ano desejado, no conjunto
referente ao indicador.

Assim, distingo a série temporal contida na linha do valor
referente a um ano específico, que também pode estar ausente.